pkg/cloudprovider/providers/azure/azure_managedDiskController.go

```diff
@@ -174,7 +174,7 @@ func (c *ManagedDiskController) getDisk(resourceGroup, diskName string) (string,
 		return *(*result.DiskProperties).ProvisioningState, *result.ID, nil
 	}
 
-	return "", "", err
+	return "", "", nil
```

Comment:
This is a case where response does not contain required property, thus should craft an explicit error here. @andyzhangx 

@tanshanshan appreciate the findings, but I'd suggest reverting this file change here.

If you would like to do a fix, I'd suppose changing err to something like
```
fmt.Errorf("Invalid response for getDisk")
```